### ISSUE TITLE
Report the usage of accessing the companion function or variable with explicit `Companion` use

### ISSUE THREAD

**Issue by atulgpt** [detekt/detekt#8926]
## Expected Behavior of the rule
```kt
// non compliant code
class A {
    companion object {
        fun foo() = 1
        val BAZ = 2
    }
}

class AFactory {
    companion object Factory {
        fun foo() = 1
    }
}

class B {
    init {
        // below are violations
        println(A.Companion.foo())
        println(AFactory.Factory.foo())
        println(A.Companion.BAZ)
    }
}
```

## Context
Found one violation in the detekt codebase https://github.com/detekt/detekt/blob/6fe62b46f3fd44ea941ecf298320de0c252dcdde/detekt-rules-style/src/main/kotlin/dev/detekt/rules/style/UnnecessaryInnerClass.kt#L60
